src/Microsoft.DotNet.Pkg.Tests/UnpackPackTests.cs

```diff
@@ -0,0 +1,243 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using FluentAssertions;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Pkg.Tests
+{
+    public class UnpackPackTests
+    {
+        private readonly ITestOutputHelper output;
+        private static readonly string simplePkg = GetResourceFilePath("Simple.pkg");
+        private static readonly string withAppPkg = GetResourceFilePath("WithApp.pkg");
+        private static readonly string simpleInstallerPkg = GetResourceFilePath("SimpleInstaller.pkg");
+        private static readonly string withAppInstallerPkg = GetResourceFilePath("WithAppInstaller.pkg");
+
+        private static readonly string pkgToolPath = Path.Combine(
+            Path.GetDirectoryName(typeof(UnpackPackTests).Assembly.Location)!,
+            "tools",
+            "pkg",
+            "Microsoft.Dotnet.Pkg.dll");
+
+        private static readonly string[] simplePkgFiles =
+        [
+            "Bom",
+            "PackageInfo",
+            Path.Combine("Payload", "Sample.txt")
+        ];
+
+        private static readonly string[] withAppPkgFiles =
+        [
+            "Bom",
+            "PackageInfo",
+            Path.Combine("Payload", "test.app")
+        ];
+
+        private static readonly string[] appFiles =
+        [
+            Path.Combine("Contents", "Info.plist"),
+            Path.Combine("Contents", "MacOS", "main"),
+            Path.Combine("Contents", "Resources", "libexample.dylib")
+        ];
+
+        private static readonly string[] simpleInstallerFiles =
+        [
+            "Distribution",
+            "Simple.pkg"
+        ];
+
+        private static readonly string[] withAppInstallerFiles =
+        [
+            "Distribution",
+            "WithApp.pkg"
+        ];
+
+        public UnpackPackTests(ITestOutputHelper output) => this.output = output;
+
+        [MacOSOnlyFact]
+        public void UnpackPackSimplePkg()
+        {
+            string unpackPath = Path.GetTempFileName();
+            string packPath = GetTempPkgPath();
+
+            ExecuteWithCleanup(() =>
+            {
+                Unpack(simplePkg, unpackPath, simplePkgFiles);
+                Pack(unpackPath, packPath, simplePkgFiles);
+            }, new List<string> { unpackPath, packPath });
```

Comment:
You can use this syntax throughout.

```suggestion
            }, [ unpackPath, packPath ]);
```